packages/matchbox/src/components/Pagination/Pagination.js

```diff
@@ -128,55 +82,111 @@ class Pagination extends Component {
     return start;
   }
 
-  render() {
-    const { pages, pageRange, marginsHidden, flat, className } = this.props;
+  React.useEffect(() => {
+    handlePageChange(currentPage - 1);
+  }, [pages, pageRange, currentPage]);
 
-    const { hasPrevious, hasNext } = this.state;
+  const start = _getStart();
 
-    const start = this._getStart();
+  const buttonMarkup = () => {
+    if (pages <= pageRange) {
+      return _createButtons(0, pages);
+    } else {
+      return _createButtons(start, start + pageRange);
+    }
+  };
 
-    const buttonMarkup = () => {
-      if (pages <= pageRange) {
-        return this._createButtons(0, pages);
-      } else {
-        return this._createButtons(start, start + pageRange);
-      }
-    };
-
-    const firstButton =
-      !marginsHidden && start > 1 ? (
-        <span>
-          <Button flat={flat} className={styles.Start} onClick={() => this.handlePageChange(0)}>
-            1
-          </Button>
-          <MoreHoriz className={styles.Ellipse} style={{ marginTop: '-0.2em' }} />
-        </span>
-      ) : null;
-
-    const lastButton =
-      !marginsHidden && start + pageRange < pages ? (
-        <span>
-          <MoreHoriz className={styles.Ellipse} style={{ marginTop: '-0.2em' }} />
-          <Button
-            flat={flat}
-            className={styles.End}
-            onClick={() => this.handlePageChange(pages - 1)}
-          >
-            {pages}
-          </Button>
-        </span>
-      ) : null;
-
-    return (
-      <div className={classnames(styles.Pagination, className)}>
-        <Pager.Previous flat={flat} onClick={this.handlePrevious} disabled={!hasPrevious} />
-        {firstButton}
-        <span className={styles.Pages}>{buttonMarkup()}</span>
-        {lastButton}
-        <Pager.Next flat={flat} onClick={this.handleNext} disabled={!hasNext} />
-      </div>
-    );
-  }
+  const firstButton =
+    !marginsHidden && start > 1 ? (
+      <span>
+        <Button
+          flat
+          width={index + 1 < 100 ? tokens.spacing_600 : 'auto'}
+          onClick={() => handlePageChange(0)}
+        >
+          1
+        </Button>
+        <Box display="inline" pl={200} pr={200}>
+          <StyledMoreHoriz size={24} />
+        </Box>
+      </span>
+    ) : null;
+
+  const lastButton =
+    !marginsHidden && start + pageRange < pages ? (
+      <span>
+        <Box display="inline" pl={200} pr={200}>
+          <StyledMoreHoriz size={24} />
+        </Box>
+        <Button
+          flat
+          width={index + 1 < 100 ? tokens.spacing_600 : 'auto'}
+          onClick={() => handlePageChange(pages - 1)}
+        >
+          {pages}
+        </Button>
+      </span>
+    ) : null;
+
+  return (
+    <Box display="inline-flex" alignItems="center" {...rest}>
```

Comment:
using Box here will allow all system-props through, not just margin. You could just add another div wrapper aound this with those system props